src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -381,8 +424,30 @@ public native Object[] newSpecialArray(Class<?> componentType,
 
     /**
      * Returns an object instance with a private buffered value whose layout
-     * and contents is exactly the given value instance.  The return object
-     * is in the larval state that can be updated using the unsafe put operation.
+     * and contents is exactly the given value instance. The return object is
+     * in the larval state that can be updated using the unsafe put operation.
+     * <p>
+     * This method mimics the operation of the bytecode {@code new}, allowing
+     * the creation of a value object without the conventional Java procedure
+     * and bypassing the bytecode verifier regarding larval objects. As a
+     * result, it is the responsibility of the caller to ensure that this
+     * method is used in a conformant manner. This method exhibits
+     * {@linkplain Unsafe undefined behavior} unless all the following
+     * conditions are true:
+     * <ul>
+     * <li>The argument must be a value object at runtime.
+     * <li>The return object must be assigned to a local variable that is
+     * effectively final as well as definitely assigned to with the return
+     * value of this method. The object must also be not assigned to another
+     * local variable.
```

Comment:
Why this restriction? What assumption is violated if there is a second assignment?